common/djangoapps/dark_lang/views.py

```diff
@@ -0,0 +1,153 @@
+"""
+Views file for the Darklang Django App
+"""
+from openedx.core.lib.api.view_utils import view_auth_classes
+from django.views.generic.base import View
+from django.contrib.auth.decorators import login_required
+from django.utils.decorators import method_decorator
+
+from edxmako.shortcuts import render_to_response
+
+from openedx.core.djangoapps.user_api.preferences.api import (
+    delete_user_preference, get_user_preference, set_user_preference
+)
+from lang_pref import LANGUAGE_KEY
+
+from django.utils.translation import LANGUAGE_SESSION_KEY
+from django.utils.translation import ugettext as _
+
+from dark_lang import DARK_LANGUAGE_KEY
+from dark_lang.models import DarkLangConfig
```

Comment:
Good question-- and I'd include in this question, how do I know what libraries are edX vs. core python or third party? I recently was trying to find documentation for request_cache.middleware, which I assumed was not our own code, but just found out it actually is part of edx-platform.

I miss Java and package naming conventions.